src/lib/oauth-client.ts

```diff
@@ -0,0 +1,73 @@
+import { AuthorizationCode } from 'simple-oauth2';
+import type { AccessToken } from 'simple-oauth2';
+
+import { Permanent } from './client';
+
+class PermanentOAuthClient {
+  private client: AuthorizationCode;
+
+  public constructor(
+    clientId: string,
+    clientSecret: string,
+    private baseUrl = 'https://www.permanent.org/api',
+    authHost = 'https://auth.permanent.org/'
+  ) {
+    this.client = new AuthorizationCode({
+      client: {
+        id: clientId,
+        secret: clientSecret,
+      },
+      auth: {
+        authorizePath: '/oauth2/authorize',
+        tokenHost: authHost,
+        tokenPath: '/oauth2/token',
+      },
+    });
+  }
+
+  public authorizeUrl(
+    callbackUrl: string,
+    scope: string,
+    state: string
+  ): string {
+    // TODO: generate CSRF token and include it in state
```

Comment:
I think PER-8678 tackles this from the other side - that is, we won't require CSRF tokens on OAuth requests.  Right?

Reply:
Ah, no, PER-8678 is about CSRF tokens in requests to Permanent; this is about securing the OAuth authorization code grant flow between the client and FusionAuth, to ensure that the client application is not attacked during the login process. I do still want to do this, and I'll find some supporting documentation when I do - all I have right now is the conclusion I remember drawing from unrelated work last spring.

Reply:
Thank you!